src/watchful.ts

```diff
@@ -120,12 +121,15 @@ export class Watchful extends Construct implements IWatchful {
     this.dash?.addWidgets(...widgets);
   }
 
-  public addAlarm(alarm: cloudwatch.Alarm) {
+  public addAlarm(alarm: cloudwatch.IAlarm) {
+    const alarmWithAction = hasAlarmAction(alarm) ? alarm : new cloudwatch.CompositeAlarm(this, `Created Alarm ${this.createdAlarmCount++}`, {
+      alarmRule: cloudwatch.AlarmRule.fromAlarm(alarm, cloudwatch.AlarmState.ALARM),
```

Comment:
Sure. To be clear the test case would be for some custom construct implementing IAlarm that lacks alarmAction?

Reply:
yes, I think [`Alarm.fromAlarmArn`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-cloudwatch.Alarm.html#static-fromwbralarmwbrarnscope-id-alarmarn) should work for this purpose

Reply:
That wouldn't work because internally `fromAlarmArn` generates a new class that extends AlarmBase, and will have addAlarmAction. I've updated the PR with a test that works against a new custom alarm class. I'm having a little trouble seeing a real-world use case for this given the internals of `fromAlarmArn`, but if you think it's best we can include it.

Reply:
Simply because our hope is that `IAlarm` encapsulates enough substance that there *could* be an implementation that is not necessarily provided by the AWS CDK construct library. Understood that there is no such implementation that we are aware of at this time